src/app/profiles/profiles.editor.modal.ts

```diff
@@ -50,131 +48,101 @@ export class ProfilesEditorModalComponent {
+    check_reply_to(identity) {
+        if (identity && identity.email && identity.reply_to &&
+            identity.reply_to !== identity.email) {
             this.is_different_reply_to = true;
             return;
         }
         this.is_different_reply_to = false;
     }
-    set_localpart(data) {
-        if (data.profile.email.match(/@/g)) {
-            this.localpart = data.profile.email.replace(/@.+/g, '');
+    set_localpart(identity) {
+        if (identity.email.match(/@/g)) {
+            this.localpart = identity.email.replace(/@.+/g, '');
             const regex = /(.+)@(.+)/g;
-            const match = regex.exec(data.profile.email);
-            data.profile.preferred_runbox_domain = match[2];
+            const match = regex.exec(identity.email);
+            identity.preferred_runbox_domain = match[2];
         } else {
-            this.localpart = data.profile.email;
-            data.profile.preferred_runbox_domain = this.localpart;
+            this.localpart = identity.email;
+            identity.preferred_runbox_domain = this.localpart;
         }
     }
     save() {
-        if (this.is_create || this.is_create_main) {
+        if (this.is_create) {
             this.create();
         } else { this.update(); }
     }
     create() {
-        this.is_busy = true;
-        const data = this.data;
+        const identity = this.identity;
         const values = {
-            name: data.profile.name,
-            email: data.profile.email,
-            from_name: data.profile.from_name,
-            reply_to: data.profile.reply_to,
-            signature: data.profile.signature,
-            smtp_address: data.profile.smtp_address,
-            smtp_port: data.profile.smtp_port,
-            smtp_username: data.profile.smtp_username,
-            smtp_password: data.profile.smtp_password,
+            name: identity.name,
+            email: identity.email,
+            from_name: identity.from_name,
+            reply_to: identity.reply_to,
+            signature: identity.signature,
+            smtp_address: identity.smtp_address,
+            smtp_port: identity.smtp_port,
+            smtp_username: identity.smtp_username,
+            smtp_password: identity.smtp_password,
             type: this.type,
-            is_signature_html: (data.profile.is_signature_html ? 1 : 0),
-            is_smtp_enabled: (data.profile.is_smtp_enabled ? 1 : 0),
+            is_signature_html: (identity.is_signature_html ? 1 : 0),
+            is_smtp_enabled: (identity.is_smtp_enabled ? 1 : 0),
         };
-        const req = this.rmm.profile.create(values, this.field_errors);
-        req.subscribe(reply => {
-            if (reply['status'] === 'success') {
-                this.rmm.profile.load();
-                this.close();
-                return;
-            }
-            this.is_busy = false;
-        });
+        this.profileService.create(values).subscribe(
+            res => this.close
```

Comment:
`this.close()`, it won't call the function otherwise (I also write too much perl 😆)